core/src/main/java/org/apache/calcite/rel/rules/AggregateReduceFunctionsRule.java

```diff
@@ -94,15 +94,23 @@
 
   /** The singleton. */
   public static final AggregateReduceFunctionsRule INSTANCE =
-      new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()),
+      new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()), true,
           RelFactories.LOGICAL_BUILDER);
 
+  /** The singleton. */
+  public static final AggregateReduceFunctionsRule NO_REDUCE_SUM =
```

Comment:
In particular project there may be a use-case that does not need to reduce sum. If necessary the singleton should be added there and not here. From my perspective it is unlikely that there will be many people who are going to use this rule.

Reply:
Rather than adding a new singleton instance, is it okay to add a new constructor to get this behavior?

Reply:
There can't be two singletons. By definition of the word 'singleton'.

Reply:
done.